### PR TITLE
Feature 16

### PR DESCRIPTION
**Pull request by xus84**
## Summary
This pull request adds support for filtering CSV and JSON files in the drag & drop component and removes an unused library.

## Changes Made
Added the ability to filter for CSV and JSON files in the drag & drop component, allowing for more efficient file handling.
Deleted an unused library that was not needed for the current implementation.

### CODE REVIEW

File: pages/api/finance/index.ts [hunk 1]
```diff
@@ -3,12 +3,37 @@ import { EasyFinance } from 'easy-personal-finance'
 import { FinanceTransaction } from '@/pages/interfaces'
 import financeData from './data/finance-transactions.json'
 import crypto from 'crypto'
+import { FinanceDataConverter } from '../../../lib/index'
```
Comment (smarcilla) [inline pages/api/finance/index.ts]: Remove this line please. Remember that you remove lib directory

File: pages/transactions.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { data } from './data/finance-transactions.json'
 import { EasyFinance } from 'easy-personal-finance'
-import { FinanceTransactionEntity } from 'easy-personal-finance/lib/entities'
+//import { FinanceTransactionEntity } from 'easy-personal-finance/lib/entities'
```
Comment (smarcilla) [inline pages/transactions.tsx]: Remove it is better than comment

**Review by smarcilla — APPROVED** [review]

File: pages/transactions.tsx (hunk 2 above)
Comment (xus84) [reply pages/transactions.tsx]: Done

File: components/FilePreview.tsx [hunk 3]
```diff
@@ -1,7 +1,16 @@
-import React from 'react'
+import React, { useState } from 'react'
```
Comment (smarcilla) [inline components/FilePreview.tsx]: You can remove {useState} because it doesn't use in FilePreview

File: package.json [hunk 4]
```diff
@@ -18,12 +18,14 @@
     "@types/react": "18.0.26",
     "@types/react-dom": "18.0.10",
     "autoprefixer": "^10.4.13",
+    "convert-csv-to-json": "^1.5.0",
```
Comment (smarcilla) [inline package.json]: you can remove this dependency.

File: pages/index.tsx [hunk 5]
```diff
@@ -92,7 +92,7 @@ export default function Index() {
           their finances.
         </p>
       </section>
-      <div className="grid grid-cols-1 p-6 md:grid-cols-4 lg:grid-cols-5">
+      {/*     <div className="grid grid-cols-1 p-6 md:grid-cols-4 lg:grid-cols-5">
```
Comment (smarcilla) [inline pages/index.tsx]: why is this code commented?

File: pages/transactions.tsx (hunk 2 above)
Comment (smarcilla) [reply pages/transactions.tsx]: Line 3 start with a comment. Can you remove `import { FinanceTransactionEntity } from 'easy-personal-finance/lib/entities'` please?

**Review by smarcilla — CHANGES_REQUESTED** [review]

**Comment by xus84** [comment]
Everything is ready.

**Review by smarcilla — APPROVED** [review]

**Pull request merged**